pi_mqtt_gpio/server.py

```diff
@@ -446,6 +477,85 @@ def initialise_digital_output(out_conf, gpio):
     gpio.setup_pin(out_conf["pin"], PinDirection.OUTPUT, None, out_conf)
 
 
+def initialise_sensor_input(sens_conf, sensor):
+    """
+    Initialises sensor input.
+    :param in_conf: Sensor config
+    :type in_conf: dict
+    :param sensor: Instance of GenericSensor to use
+    :type sensor: pi_mqtt_gpio.modules.GenericSensor
+    :return: None
+    :rtype: NoneType
+    """
+    sensor.setup_sensor(sens_conf)
+
+
+def sensor_timer_thread(SENSOR_MODULES, sensor_inputs, topic_prefix):
+    """
+    Timer thread for the sensors
+    To reduce cpu usage, there is only one cyclic thread for all sensors.
+    At the beginning the cycle time is calculated (ggT) to match all intervals.
+    For each sensor interval, the reduction value is calculated, that triggers
+    the read, round and publish for the sensor, when loop_count is a multiple
+    of it. In worst case, the cycle_time is 1 second, in best case, e.g., when
+    there is only one sensor, cycle_time is its interval.
+    """
+
+    # calculate the min time
+    arr = []
+    for sens_conf in sensor_inputs:
+        arr.append(sens_conf.get("interval", 60))
+
+    # get the greates common divisor (gcd) for the list of interval times
+    cycle_time = reduce(lambda x, y: gcd(x, y), arr)
+
+    _LOG.info(
```

Comment:
This might be a bit noisy at info level - suggest setting it to debug.
```suggestion
    _LOG.debug(
```